extensions/auth/portability-auth-microsoft/src/main/java/org/dataportabilityproject/auth/microsoft/MicrosoftAuthDataGenerator.java

```diff
@@ -45,26 +68,34 @@
    * @param clientSecretSupplier The application secret that was created in the app registration
    *     portal for the portability instance
    * @param mapper the mapper for deserializing responses from JSON
+   * @param transferDataType the data type to create this authorization generator for
+   * @param mode the mode to create this authorization generator for
    */
   public MicrosoftAuthDataGenerator(
       String redirectPath,
       Supplier<String> clientIdSupplier,
       Supplier<String> clientSecretSupplier,
       OkHttpClient client,
-      ObjectMapper mapper) {
+      ObjectMapper mapper,
+      String transferDataType,
+      AuthMode mode) {
     this.redirectPath = redirectPath;
     this.clientIdSupplier = clientIdSupplier;
     this.clientSecretSupplier = clientSecretSupplier;
     httpClient = client;
     this.mapper = mapper;
+    this.scopes =
+        mode == AuthMode.EXPORT
```

Comment:
Do we have to throw an error of it ends up null or do a precondtion check?

Reply:
Done - Presumably this shouldnt ever happen since only the AuthServiceExtension creates these, but added it as a catch.